src/components/App.js

```diff
@@ -3,20 +3,45 @@
 /* eslint-disable react/jsx-boolean-value */
 import React from 'react';
 import './App.css';
-import { Route, Switch } from 'react-router-dom';
+import { useQuery } from '@apollo/client';
+import { Switch, Route } from 'react-router-dom';
 import Homepage from './homepage/Homepage';
 import ProfessionPage from './profession-page/ProfessionPage';
 import PageNotFound from './page-not-found/PageNotFound';
 import pageTitles from '../arrays/projects-page-titles'; // import titles to be used in profession page
+import { QUERY } from '../utils/api';
 
 function App() {
   const [isProfessionPageFocused, setIsProfessionPageFocused] = React.useState(false);
   const [isMenuOpen, setIsMenuOpen] = React.useState(false);
 
+  const [isModalOpen, setIsModalOpen] = React.useState(false);
+
+  // this variable is fed into the query string in order to search the database for course project lists
+  // when updated, this will trigger a new query and update the visible list of projects
+  const [courseQuery, setCourseQuery] = React.useState('Web');
+  // ApolloClient returns from each search some data and a loading boolean that is true if the fetch is incomplete
+  // when the fetch is complete and data is populated, loading is then false
+  const { data, loading } = useQuery(QUERY, {
+    variables: { classList: courseQuery },
+  });
+
   function handleNavClick() {
     setIsMenuOpen(!isMenuOpen);
   }
 
+  function handleCourseClick(queryString) {
+    setCourseQuery(queryString);
+  }
+
+  function handleButtonClick() {
```

Comment:
Good you moved this to App.js and connected it to both Homepage and Profession Page!